### PR TITLE
French Parsing

### PR DESCRIPTION
**Pull request by GiroudMathias**
Update French Parsing

### CODE REVIEW

File: src/NzbDrone.Core/Parser/Parser.cs [hunk 1]
```diff
@@ -240,7 +240,7 @@ public static class Parser
         private static readonly Regex AnimeReleaseGroupRegex = new Regex(@"^(?:\[(?<subgroup>(?!\s).+?(?<!\s))\](?:_|-|\s|\.)?)",
                                                                 RegexOptions.IgnoreCase | RegexOptions.Compiled);
 
-        private static readonly Regex LanguageRegex = new Regex(@"(?:\W|_)(?<italian>\b(?:ita|italian)\b)|(?<german>german\b|videomann)|(?<flemish>flemish)|(?<greek>greek)|(?<french>(?:\W|_)(?:FR|VOSTFR)(?:\W|_))|(?<russian>\brus\b)|(?<dutch>nl\W?subs?)|(?<hungarian>\b(?:HUNDUB|HUN)\b)",
+        private static readonly Regex LanguageRegex = new Regex(@"(?:\W|_)(?<italian>\b(?:ita|italian)\b)|(?<german>german\b|videomann)|(?<flemish>flemish)|(?<greek>greek)|(?<french>(?:\W|_)(?:FRENCH|VFF|VFQ)(?:\W|_))|(?<russian>\brus\b)|(?<dutch>nl\W?subs?)|(?<hungarian>\b(?:HUNDUB|HUN)\b)",
```
Comment (markus101) [inline src/NzbDrone.Core/Parser/Parser.cs]: French is already handled below.

File: src/NzbDrone.Core/Parser/Parser.cs [hunk 2]
```diff
@@ -240,7 +240,7 @@ public static class Parser
         private static readonly Regex AnimeReleaseGroupRegex = new Regex(@"^(?:\[(?<subgroup>(?!\s).+?(?<!\s))\](?:_|-|\s|\.)?)",
                                                                 RegexOptions.IgnoreCase | RegexOptions.Compiled);
 
-        private static readonly Regex LanguageRegex = new Regex(@"(?:\W|_)(?<italian>\b(?:ita|italian)\b)|(?<german>german\b|videomann)|(?<flemish>flemish)|(?<greek>greek)|(?<french>(?:\W|_)(?:FR|VOSTFR)(?:\W|_))|(?<russian>\brus\b)|(?<dutch>nl\W?subs?)|(?<hungarian>\b(?:HUNDUB|HUN)\b)",
```
Comment (markus101) [inline src/NzbDrone.Core/Parser/Parser.cs]: Why are `FR` and `VOSTFR` being removed?

**Comment by markus101** [comment]
Tests for these need to be added to the `LanguageParserFixture`.

File: src/NzbDrone.Core/Parser/Parser.cs (hunk 2 above)
Comment (GiroudMathias) [inline src/NzbDrone.Core/Parser/Parser.cs]: the french torrents are generally written "FRENCH, VFF VFQ".

Leaving the VOSTFR and GB, the files are downloaded in English and therefore not in french

**Comment by Twanislas** [comment]
Hi !

Please see my comment in #280 ;)

Would'nt an even better regexp be : `(?<french>(?:\W|_)(?:fr|french|vf|truefrench|vff|vfq)(?:\W|_))` ?

Cheers !

**Comment by JigSawFr** [comment]
hm, why removing VOSTFR ? I'm using it to download all serie's episodes...
if not available in french, it use VOSTFR actually

**Comment by Twanislas** [comment]
Hi @JigSawFr !

Actually I see this more as a bug than a feature. While it's nice to have it "magically" fallback to VOSTFR, some people may want to download a series only in french.

Moreover, this could end up in a mess with one episode on 2 being VOSTFR and the other in VF. Those of us who are OCD'ed about having a nice and perfeclty organized library would be very upset :D

Yet one could add a tag to a series to exclude the VOSTFR word from search results but then it would mean overriding the behavior we expect when we instruct the software to download things in french.

As I see it, if I put the language setting to "French", that's what I expect Sonarr to download. If I wanted VOSTFR, I'd set it to english, and add a filter. Or try to add another language in Sonarr for that (but that may be overkill ?)

Cheers !

**Comment by Twanislas** [comment]
Sent PR #1389.

Cheers :) !

**Comment by Taloth** [comment]
@GiroudMathias @Twanislas I'm gonna respond to both your PRs right here. As it stands, neither are acceptable to us.

The problem is that you may not consider VOSTFR to be French, but it basically is. Just like NLSub is Dutch. Both may be 'just' subtitles, but the majority of users don't want those.
It's notoriously hard to separate video/audio language and subtitles.
Also, early NLSub releases actually were hardcoded subs, not embedded srts.

Removing those from the language detection basically forces everyone to add Restrictions, which kinda  defeats the goal of Sonarr aiming to be easy to use for 90% of the users.

So, lets think about a solution.
What we need to do is at least be able to make a distinction which language tags are video/audio and which are guaranteed subtitle tags. That's relatively easy, just create separate regexes.
Then we need to come up with a way to let the user decide whether he's okay with foreign sub tags.
Best place to do that atm is the Profile, but an global flag might be an option as well.

We had a PR that created separate Language profiles and multi-lang support, but that's gone (the submitter account is gone).
Sonarr doesn't support foreign language metadata atm, so there's little point in going all the way.

**Comment by JigSawFr** [comment]
@Taloth 
Best way in my mind is possibility to download in French and fallback to subtitles (VOSTFR) if no releases found. And why not to English if nothing found. Maybe an order sytem for languages

**Comment by Twanislas** [comment]
Hi !

While I like you idea @JigSawFr, how would you prevent episodes language mixup in the series ?

Actually, a language order system sounds super nice. Kinda the same system as we already have for qualities, where we can set a cutoff. So we can have early releases in VOSTFR which will be updated when the french version shows up (or not, depending on cutoff point). Me likey !

Tha would mean I don't have to lookup every show in my library to know wether they air in french or english and set tags/filters/restrictions accordingly show by show.

Cheers !

**Comment by markus101** [comment]
Closing, but we're open for more discussion on this if anyone wishes to  start dialog there, please open a GHI.

**Pull request closed without merging**